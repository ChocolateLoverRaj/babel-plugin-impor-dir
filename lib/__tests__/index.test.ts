import plugin = require('../index')
import never = require('never')
import { transformFileSync } from '@babel/core'
import { join } from 'path'

const resDir = join(__dirname, './res')

const snapshotFile = (file: string): void => {
  const { code } = transformFileSync(join(resDir, file), {
    plugins: [plugin]
  }) ?? never()
  expect(code).toMatchSnapshot()
}

test('named imports', () => snapshotFile('namedImports.js'))

test('default import', () => snapshotFile('defaultImport.js'))

test('import file', () => snapshotFile('importFile.js'))
