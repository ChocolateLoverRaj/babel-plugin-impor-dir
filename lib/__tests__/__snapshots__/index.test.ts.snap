// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`default import 1`] = `
"// eslint-disable-next-line no-unused-vars
import _utilGoodNumber from \\"./util/goodNumber.js\\";
import _utilIsOdd from \\"./util/isOdd.js\\";
const util = {
  \\"goodNumber\\": _utilGoodNumber,
  \\"isOdd\\": _utilIsOdd
};"
`;

exports[`import file 1`] = `
"// eslint-disable-next-line no-unused-vars
import isOdd from './util/isOdd';"
`;

exports[`named imports 1`] = `
"// eslint-disable-next-line no-unused-vars
import goodNumber from \\"./util/goodNumber\\";
import isOdd from \\"./util/isOdd\\";"
`;
